pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -81,6 +98,20 @@ func (configgen *ConfigGeneratorImpl) buildGatewayListeners(env model.Environmen
 			log.Warnf("Multiple servers on same port is not supported yet, port %d", server.Port.Number)
 			continue
 		}
+		if server.Tls != nil {
+			if server.Tls.Mode == networking.Server_TLSOptions_MUTUAL && server.Tls.CaCertificates == "" {
```

Comment:
this should be caught by validation. Not here.

Reply:
Agreed - can cleanup later, but 2 checks better than none :-)

Reply:
This already exists: https://github.com/istio/istio/blob/654617da2c7f12a6e5fc97a1f2779f1cae23cb4f/pilot/pkg/model/validation.go#L947-L954

If all that's missing is the check for the private key's value, then please add it there and delete this code. I don't want validation code mixed up in this business logic, it just complicates things down the road.